aws_signing_helper/update.go

```diff
@@ -98,90 +98,92 @@ func GetCredentialsFileContents() ([]string, error) {
 }
 
 // Assume that the credentials file exists already and open it for write operations
+// that will overwrite the existing contents of the file
 func GetWriteOnlyCredentialsFile() (*os.File, error) {
```

Comment:
This should be refactored in a way to make it easier to unit test. For example - over-writing in place makes it hard to compare original and modified copies. Or, you could separate the logic that builds the file in memory and use that for comparison.

Reply:
Sure, I've fixed it in the next revision.